From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Thu, 17 Aug 2023 23:42:02 +0200
Subject: init: Include the error stack in the context termination error
 message

If a failure happened during initialization the shell does not provide
any debug information, and so only the error is shown without a stack
trace.

Since this information is provided, pass it as the error message.

Do not log this directly from JS so that we just use one termination
path.

Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2899>
(cherry picked from commit 49464e3c7ab88f07d12d8f544251acd21ecf6551)
---
 js/ui/init.js | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/js/ui/init.js b/js/ui/init.js
index 2dea981..a7cf7db 100644
--- a/js/ui/init.js
+++ b/js/ui/init.js
@@ -11,7 +11,7 @@ imports._promiseNative.setMainLoopHook(() => {
         import('./main.js').then(main => main.start()).catch(e => {
             const error = new GLib.Error(
                 Gio.IOErrorEnum, Gio.IOErrorEnum.FAILED,
-                e.message);
+                `${e.message}\n${e.stack}`);
             global.context.terminate_with_error(error);
         });
         return GLib.SOURCE_REMOVE;
